### PR TITLE
🪱 Bug: Auth Destructuring Error

### PR DESCRIPTION
**Pull request by demariadaniel**
Addresses issue w/ `Cannot destructure property 'auth' of 'e' as it is undefined` error in browser

### CODE REVIEW

File: src/lib/gql.ts [hunk 1]
```diff
@@ -55,7 +55,7 @@ export const createApolloClient = (config: Config) => {
 	const authLink = new ApolloLink((operation, forward) => {
 		operation.setContext(({ headers }: { headers: any }) => ({
 			headers: {
-				authorization: `Bearer ${jwt}`,
+				authorization: jwt ? `Bearer ${jwt}` : '',
```
Comment (joneubank) [inline src/lib/gql.ts:58]: Perhaps we shouldnt be setting an authorization header if we have no token?

File: src/app/hooks/ApolloProvider.tsx [hunk 2]
```diff
@@ -29,7 +29,7 @@ export const ApolloProvider = ({ children }: { children: ReactNode }) => {
 	const auth = useAuthContext();
 	const { GATEWAY_API_ROOT } = useAppConfigContext();
 	const config = {
-		jwt: auth.egoJwt,
+		jwt: auth?.egoJwt,
```
Comment (joneubank) [inline src/app/hooks/ApolloProvider.tsx:32]: `useAuthContext ` should never return as undefined, the type is well defined and provided with a good default value. This `?` shouldn't be necessary.

File: src/lib/gql.ts (hunk 1 above)
Comment (demariadaniel) [reply src/lib/gql.ts:58]: Sure

```
	const authHeader = jwt ? {authorization: `Bearer ${jwt}`} : {};	

	const authLink = new ApolloLink((operation, forward) => {
		operation.setContext(({ headers }: { headers: any }) => ({
			headers: {
				...authHeader,
				...headers,
			},
```

File: src/app/hooks/ApolloProvider.tsx (hunk 2 above)
Comment (demariadaniel) [reply src/app/hooks/ApolloProvider.tsx:32]: OK fair this was a misfire, this is the only key `auth` in our app, but the error is key `auth` in a dependency
Reverted, but don't think the Headers issue above is enough to explain the bug either... investigation continues

File: src/lib/gql.ts (hunk 1 above)
Comment (justincorrigible) [reply src/lib/gql.ts]: Not asking for change: this has lower cognitive burden.
```
const authHeader = jwt && {authorization: `Bearer ${jwt}\`};
```

Comment (joneubank) [reply src/lib/gql.ts]: Doesn't that break the next section's `...authHeader` spread?

Comment (justincorrigible) [reply src/lib/gql.ts]: nup. try it out ;) 

`{ ...false } == { }`

Comment (demariadaniel) [reply src/lib/gql.ts]: Weird!

**Pull request closed without merging**